### ISSUE TITLE
Handle markers properly

### ISSUE THREAD

**Issue by jackkamm** [jackkamm/undo-propose-el#22]
PR https://github.com/jackkamm/undo-propose-el/pull/21 by @takaxp fixes a problem with `org-clock-marker`, but this problem is a symptom of a deeper problem, that we don't currently handle markers in the undo-propose buffers.

### References
- [Elisp manual: Overview of markers](https://www.gnu.org/software/emacs/manual/html_node/elisp/Overview-of-Markers.html#Overview-of-Markers)
- [Elisp manual: Undo](https://www.gnu.org/software/emacs/manual/html_node/elisp/Undo.html), see the entry on the marker element
- `primitive-undo` in simple.el, in particular [these lines](https://github.com/emacs-mirror/emacs/blob/13b951001c15a78f7f8cb4bff1825cc77b2c8456/lisp/simple.el#L2606-L2628) which handle marker adjustments. 

### Additional Notes
There are entries in the undo-list which adjust marker positions. When in the undo-propose buffer, `(eq (marker-buffer m) (current-buffer))` is false so these entries are skipped.

One possible way to handle markers would be to copy the markers over to the undo-propose buffer, and edit entries in the undo-list to use the new markers.

**Comment by jackkamm** [comment]
It looks like grabbing and copying all the markers is not possible:
- https://stackoverflow.com/questions/36926513/are-there-elisp-functions-that-list-markers-in-a-given-buffer
- https://groups.google.com/forum/#!topic/gnu.emacs.help/vNG9y5qCaxc/discussion

**Comment by jackkamm** [comment]
As a workaround until we've found a permanent solution, we could create a list `undo-propose--markers-to-check` that we could add markers to (e.g. `org-clock-marker` from #21), and undo-propose will check for those markers and make copies to keep track of & update them.

**Comment by takaxp** [comment]
Yes. The important thing is that we actually don't need to handle all markers in a buffer. It is also hard to identify which markers should be checked by us for all existing packages. I think it is a good choice that only registered markers will be handled properly.

In the case of `org-clock-marker`, we don't need to always copy the marker and also the original maker in the original buffer shouldn't be touched in the case of calling `undo-propose-cancel`.

In my understanding, the markers will not be actually copied but **forked** from the original marker for editing works in `undo-propose` buffer. If we commit the buffer with some undos, then the original marker shall be updated (normally updating the position). If we don't commit, then **forked** marker will be just trashed.
